streaming/java/streaming-state/src/main/java/org/ray/streaming/state/keystate/state/facade/MapStateFacade.java

```diff
@@ -0,0 +1,41 @@
+package org.ray.streaming.state.keystate.state.facade;
+
+import java.util.Map;
+import org.ray.streaming.state.IKVState;
+import org.ray.streaming.state.backend.KeyStateBackend;
+import org.ray.streaming.state.keystate.desc.MapStateDescriptor;
+import org.ray.streaming.state.keystate.state.MapState;
+import org.ray.streaming.state.keystate.state.impl.MapStateImpl;
+import org.ray.streaming.state.strategy.AbstractTransactionState;
+
+/**
+ * This class defines MapState Wrapper, connecting state and backend.
+ */
+public class MapStateFacade<K, V> extends AbstractTransactionState<Map<K, V>> implements
+    IKVState<String, Map<K, V>> {
+
+  private KeyStateBackend keyStateBackend;
+  private MapStateImpl<K, V> mapState;
+
+  public MapStateFacade(KeyStateBackend keyStateBackend, MapStateDescriptor<K, V> stateDescriptor) {
+    super(keyStateBackend.getBackStorage(stateDescriptor), keyStateBackend.getStorageMode());
+    this.keyStateBackend = keyStateBackend;
+    this.mapState = new MapStateImpl<>(stateDescriptor, keyStateBackend);
+  }
+
+  public MapState<K, V> getMapState() {
+    return this.mapState;
+  }
+
+  @Override
+  public Map<K, V> get(String key) {
+    this.stateStrategy.setKeyGroupIndex(keyStateBackend.getKeyGroupIndex());
```

Comment:
ditto to @JingGe 's comments above. I feel this like something should be stay in the super class but we'd better factor it out. Why does the state strategy have to reset key group index on every operation?

Reply:
If we have to, we should add a method in super class like

```java
T applyOperation(Callable<T> op) {
  // why do you need the next line
  this.stateStrategy.setKeyGroupIndex(keyStateBackend.getKeyGroupIndex());

  return op.call(); // null in Void if `void` or override this method accept Runnable
}
```

Reply:
Thanks for the comment. A thread may contains multi key group Indexes, and we need a way to figure out the keygroupindex of input key. And adding a method in super class is a better way to implement the function. I'll fix this.